core/server/services/themes/middleware.js

```diff
@@ -38,31 +37,18 @@ themeMiddleware.ensureActiveTheme = function ensureActiveTheme(req, res, next) {
     }
 
     next();
-};
+}
 
-// ### Update Template Data
-// Updates handlebars with the contextual data for the current request
-themeMiddleware.updateTemplateData = function updateTemplateData(req, res, next) {
+function updateGlobalTemplateOptions(req, res, next) {
     // Static information, same for every request unless the settings change
     // @TODO: bind this once and then update based on events?
     // @TODO: decouple theme layer from settings cache using the Content API
-    var siteData = settingsCache.getPublic(),
-        labsData = _.cloneDeep(settingsCache.get('labs')),
-        themeData = {};
-
-    if (activeTheme.get()) {
```

Comment:
No need for this because this request handler runs after a handler which errors if `activeTheme.get()` is falsy

Reply:
Maybe then `ensureActiveTheme` should be called from within this middleware? :thinking:

Reply:
I'm happy to trust the sequence for now, as that is what we export and now with the tests update, also what we test :shrimp: